File: background-tasks-java/app/src/main/java/com/android/backgroundtask/main/MainActivity.java
package com.android.backgroundtask.main;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.View;

import com.android.backgroundtask.R;
import com.android.backgroundtask.asynctask.AsyncTaskFragment;
import com.android.backgroundtask.asynctaskloader.AsyncTaskLoaderFragment;
import com.android.backgroundtask.broadcasts.BroadcastReceiverFragment;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.buttonAsyncTask).setOnClickListener(this);
        findViewById(R.id.buttonAsyncTaskLoader).setOnClickListener(this);
        findViewById(R.id.buttonCustomBroadcastReceiver).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.buttonAsyncTask) {
            loadFragment(AsyncTaskFragment.newInstance());
        }
        if (v.getId() == R.id.buttonAsyncTaskLoader) {
            loadFragment(AsyncTaskLoaderFragment.newInstance());
        }
        if (v.getId() == R.id.buttonCustomBroadcastReceiver) {
            loadFragment(BroadcastReceiverFragment.newInstance());
        }
    }

    private void loadFragment(Fragment fragment) {
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction transaction = fm.beginTransaction();
        transaction.replace(R.id.fragmentLayout, fragment);
        transaction.commit();
    }
}

File: data-binding-kot/app/src/main/java/com/android/databinding/ui/activity/MainActivity.kt
package com.android.databinding.ui.activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.android.databinding.R
import com.android.databinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    // Obtain ViewModel from ViewModelProviders, the viewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.observableFieldsActivityButton.setOnClickListener {
            startActivity(Intent(this, ObservableFieldActivity::class.java))
        }
        binding.viewmodelActivityButton.setOnClickListener {
            startActivity(Intent(this, ViewModelActivity::class.java))
        }
        binding.twoWayBindingActivityButton.setOnClickListener {
            startActivity(Intent(this, TwoWayBindingActivity::class.java))
        }
    }
}
